src/Network/Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorObject.cs

```diff
@@ -0,0 +1,311 @@
+        public string Notes { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            Validate();
+
+            PSNetworkWatcherConnectionMonitorObject CMObject = new PSNetworkWatcherConnectionMonitorObject()
+            {
+                NetworkWatcherName = this.NetworkWatcherName,
+                ResourceGroupName = this.ResourceGroupName,
+                Name = this.Name
+            };
+
+            if (ParameterSetName.Contains("SetByResource"))
+            {
+                CMObject.NetworkWatcherName = this.NetworkWatcher.Name;
+                CMObject.ResourceGroupName = this.NetworkWatcher.ResourceGroupName;
+                CMObject.Location = this.NetworkWatcher.Location;
+            }
+            else if (ParameterSetName.Contains("SetByName"))
+            {
+                MNM.NetworkWatcher networkWatcher = this.NetworkClient.NetworkManagementClient.NetworkWatchers.Get(this.ResourceGroupName, this.NetworkWatcherName);
+                CMObject.Location = networkWatcher.Location;
+            }
+            else if (ParameterSetName.Contains("SetByLocation"))
+            {
+                var networkWatcher = this.GetNetworkWatcherByLocation(this.Location);
+
+                if (networkWatcher == null)
+                {
+                    throw new ArgumentException("There is no network watcher in location {0}", this.Location);
+                }
+
+                CMObject.ResourceGroupName = NetworkBaseCmdlet.GetResourceGroup(networkWatcher.Id);
+                CMObject.NetworkWatcherName = networkWatcher.Name;
+                CMObject.Location = this.Location;
+            }
+
+            if (this.TestGroup != null)
+            {
+                CMObject.TestGroup = this.TestGroup;
+            }
+
+            if (this.Output != null)
+            {
+                CMObject.Output = this.Output;
+            }
+
+            if (this.Notes != null)
+            {
+                CMObject.Notes = this.Notes;
+            }
+
+            WriteObject(CMObject);
+        }
+
+        public bool Validate()
+        {
+
+            if (this.TestGroup != null)
+            {
+                foreach (PSNetworkWatcherConnectionMonitorTestGroupObject TestGroup in this.TestGroup)
+                {
+                    // validate mandatory parameters
+                    if (string.IsNullOrEmpty(TestGroup.Name) || TestGroup.Sources == null || TestGroup.Destinations == null || TestGroup.TestConfigurations == null)
+                    {
+                        throw new ArgumentException("Test group is missing one or more mandatory parameter");
+                    }
+
+                    // validate Source and Destination Endpoints
+                    List<PSNetworkWatcherConnectionMonitorEndpointObject> Endpoints = TestGroup.Sources;
+                    Endpoints.Concat(TestGroup.Destinations);
+
+                    if (Endpoints.Any())
```

Comment:
Can this logic for validating endpoints be reused?

Reply:
done